webapp/dojo_news/src/App.spec.js

```diff
@@ -0,0 +1,221 @@
+                    {id: 1, title: "Linux", votes: 0},
+                    {id: 2, title: "Windows", votes: 0}
+                ]
+            }
+        });
+        expect(wrapper.find("#newslist").html()).not.toBe('<div id="newslist"></div>');
+
+        expect(wrapper.findAll(".newslistitem").length).toBe(3);
+    });
+    it('renders items correctly dependent on mutations of the list of NewsItems', async () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 0},
+                ]
+            }
+        });
+        expect(wrapper.find("#newslist").html()).not.toBe('<div id="newslist"></div>');
+        expect(wrapper.findAll(".newslistitem").length).toBe(1);
+
+        wrapper.vm.removeNewsListItem(0);
+
+        await Vue.nextTick();
+
+        expect(wrapper.find("#newslist").html()).toBe('<div id="newslist"></div>');
+        expect(wrapper.findAll(".newslistitem").length).toBe(0);
+
+        wrapper.vm.createNewsListItem("Test");
+        wrapper.vm.createNewsListItem("Test2");
+
+        await Vue.nextTick();
+
+        expect(wrapper.find("#newslist").html()).not.toBe('<div id="newslist"></div>');
+        expect(wrapper.findAll(".newslistitem").length).toBe(2);
+    });
+});
+
+describe("List Empty Message", () => {
+    const listEmptyMessage = "Oh noes, the list is empty!";
+
+    it('displays a message on an initially empty list of NewsItems', () => {
+        const wrapperWithEmptyList = shallowMount(App, {
+            propsData: {
+                listEmptyMessage: listEmptyMessage,
+                initialNewsListItems: []
+            }
+        });
+        expect(wrapperWithEmptyList.text()).toContain(listEmptyMessage);
+    });
+
+    it('displays no message on an initially non-empty list of NewsItems', () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                listEmptyMessage: listEmptyMessage,
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 0},
+                    {id: 1, title: "Linux", votes: 0},
+                    {id: 2, title: "Windows", votes: 0}
+                ]
+            }
+        });
+
+        expect(wrapper.text()).not.toContain(listEmptyMessage);
+    });
+
+    it('adds and removes the message on list mutations', async () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                listEmptyMessage: listEmptyMessage,
+                initialNewsListItems: []
+            }
+        });
+
+        expect(wrapper.text()).toContain(listEmptyMessage);
+
+        wrapper.vm.createNewsListItem("Test")
+
+        await Vue.nextTick();
+
+        expect(wrapper.text()).not.toContain(listEmptyMessage);
+
+        wrapper.vm.removeNewsListItem(0);
+
+        await Vue.nextTick();
+
+        expect(wrapper.html()).toContain(listEmptyMessage);
+    });
+});
```

Comment:
If your test case description already has `do X and Y` then probably you should split the test case.

Reply:
Fixed!